lib/shopify_app/logger.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+module ShopifyApp
+  module Logger
+    LOG_LEVELS = { debug: 0, info: 1, warn: 2, error: 3, off: 4 }
+    PREFIX = "ShopifyApp"
+
+    def self.send_to_logger(log_level, message)
+      return unless enabled_for_log_level?(log_level)
+
+      current_shop = ShopifyAPI::Context.active_session&.shop || "Shop Not Found"
+      message_context = "[ #{PREFIX} | #{log_level.to_s.upcase} | #{current_shop} ] #{message}"
+
+      ShopifyAPI::Context.logger.send(log_level, message_context)
+    end
+
+    def self.debug(message)
+      send_to_logger(:debug, message)
+    end
+
+    def self.info(message)
+      send_to_logger(:info, message)
+    end
+
+    def self.warn(message)
+      send_to_logger(:warn, message)
+    end
+
+    def self.error(message)
+      send_to_logger(:error, message)
+    end
+
+    def self.enabled_for_log_level?(log_level)
+      LOG_LEVELS[log_level] >= LOG_LEVELS[ShopifyApp.configuration.log_level]
```

Comment:
If the `shopify_app.rb` is missing the `log_level` setting (e.g. for someone upgrading) then this fails with:

```
Completed 500 Internal Server Error in 1ms (ActiveRecord: 0.0ms | Allocations: 1214)


  
ArgumentError (comparison of Integer with nil failed):
  
/Users/andyw8/src/github.com/Shopify/shopify_app/lib/shopify_app/logger.rb:38:in `>='
/Users/andyw8/src/github.com/Shopify/shopify_app/lib/shopify_app/logger.rb:38:in `enabled_for_log_level?'
/Users/andyw8/src/github.com/Shopify/shopify_app/lib/shopify_app/logger.rb:9:in `send_to_logger'
/Users/andyw8/src/github.com/Shopify/shopify_app/lib/shopify_app/logger.rb:18:i
```